src/doc/src/reference/config.md

```diff
@@ -415,10 +415,14 @@ wrapper is the path to the actual executable to use
 * Default: none
 * Environment: `CARGO_BUILD_RUSTC_WORKSPACE_WRAPPER` or `RUSTC_WORKSPACE_WRAPPER`
 
-Sets a wrapper to execute instead of `rustc`, for workspace members only.
-The first argument passed to the wrapper is the path to the actual
-executable to use (i.e., `build.rustc`, if that is set, or `"rustc"` otherwise).
-It affects the filename hash so that artifacts produced by the wrapper are cached separately.
+Sets a wrapper to execute instead of `rustc`, for workspace members only. When building a
+single-package project without workspaces, that package is considered to be the workspace. The first
+argument passed to the wrapper is the path to the actual executable to use (i.e., `build.rustc`, if
+that is set, or `"rustc"` otherwise). It affects the filename hash so that artifacts produced by the
+wrapper are cached separately.
+
+If both `rustc-wrapper` and `rustc-workspace-wrapper` are set, then they will be nested:
+the final invocation is `$RUSTC_WRAPPER $RUSTC_WORKSPACE_WRAPPER $RUSTC`.
```

Comment:
I hope that's right. It is what autocfg does since https://github.com/cuviper/autocfg/pull/56, but I did not verify whether it actually matches what cargo does.
Cc @cuviper

Reply:
I verified by trial and by inspection -- Cargo pushes the workspace wrapper first:
https://github.com/rust-lang/cargo/blob/5b2b1c036c6665f842148af9da10cc1a718d40c7/src/cargo/util/rustc.rs#L123-L125

and then applies the wrappers in reverse:
https://github.com/rust-lang/cargo/blob/5b2b1c036c6665f842148af9da10cc1a718d40c7/crates/cargo-util/src/process_builder.rs#L480-L485

Reply:
There is no test verifying this precedence. I've got one. Could you add it to below this [`rustc_wrapper`](https://github.com/rust-lang/cargo/blob/3b43aa875fa262f53bdcc5ad4c0f677802630186/tests/testsuite/build.rs#L5195) test?

```rust
#[cargo_test]
fn rustc_wrapper_precendence() {
    let p = project().file("src/lib.rs", "").build();
    let rustc_wrapper = tools::echo_wrapper();
    let ws_wrapper = rustc_wrapper.with_file_name("rustc-ws-wrapper");
    assert_ne!(rustc_wrapper, ws_wrapper);
    std::fs::hard_link(&rustc_wrapper, &ws_wrapper).unwrap();

    let running = format!(
        "[RUNNING] `{} {} rustc --crate-name foo [..]",
        rustc_wrapper.display(),
        ws_wrapper.display(),
    );
    p.cargo("build -v")
        .env("RUSTC_WRAPPER", &rustc_wrapper)
        .env("RUSTC_WORKSPACE_WRAPPER", &ws_wrapper)
        .with_stderr_contains(running)
        .run();
}
```